libmultilabel/metrics.py

```diff
@@ -31,68 +26,75 @@ class MultiLabelMetrics():
     def __init__(self, config):
         self.monitor_metrics = config.get('monitor_metrics', [])
         self.metric_threshold = config.get('metric_threshold', 0.5)
-        self.y_true = []
-        self.y_pred = []
-        self.cached_results = {}
+
+        self.n_eval = 0
+        self.multilabel_confusion_matrix = 0.
+        self.metric_stats = {}
 
         self.top_ks = set()
+        self.prec_recall_metrics = []
         for metric in self.monitor_metrics:
             if re.match('[P|R]@\d+', metric):
                 top_k = int(metric[2:])
                 self.top_ks.add(top_k)
+                self.metric_stats[metric] = 0.
+                self.prec_recall_metrics.append(metric)
             elif metric not in ['Micro-Precision', 'Micro-Recall', 'Micro-F1', 'Macro-F1', 'Another-Macro-F1']:
                 raise ValueError(f'Invalid metric: {metric}')
 
+    def reset(self):
+        self.n_eval = 0
+        self.multilabel_confusion_matrix = 0.
+        for metric in self.metric_stats:
+            self.metric_stats[metric] = 0.
+
     def add_values(self, y_true, y_pred):
```

Comment:
I think this can be called add_batch_metrics

Reply:
I just found people making Pytorch Lightning also have a package TorchMetrics.
They use `update()` and `computation()`.
https://torchmetrics.readthedocs.io/en/latest/pages/implement.html
Considering their popularity, I think we can follow their design?